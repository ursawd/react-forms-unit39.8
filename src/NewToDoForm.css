.input-btn {
  margin-left: 5px;
  background-color: steelblue;
  color: white;
}
