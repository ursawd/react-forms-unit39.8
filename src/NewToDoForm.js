//Add new ToDo form component
//
import { useState } from "react";
import "./NewToDoForm.css";
//------------------------------------------------------------
const NewToDoForm = ({ addToDo }) => {
  //------------------------------------------------------------
  const INITIAL_STATE = { task: "" };
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleSubmit = (evt) => {
    evt.preventDefault();
    if (formData.task === "") return;
    addToDo({ ...formData });
    setFormData(INITIAL_STATE);
  };
  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((formData) => ({ ...formData, [name]: value }));
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        onChange={handleChange}
        type="Text"
        value={formData.task}
        name="task"
        id="task"
        placeholder="Enter new ToDo here..."
      />
      <button className="input-btn" type="submit">
        Add New ToDo
      </button>
    </form>
  );
};
export default NewToDoForm;
