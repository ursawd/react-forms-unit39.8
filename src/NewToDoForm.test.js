import React from "react";
import { render, fireEvent, queryByText } from "@testing-library/react";
import NewToDoForm from "./NewToDoForm";

//smoke test
test("renders without crashing", function () {
  render(<NewToDoForm />);
});

//snapshot
test("matches snapshot", function () {
  const { asFragment } = render(<NewToDoForm />);
  expect(asFragment()).toMatchSnapshot();
});

test("should add new Todo", function () {
  const { queryByText, queryByPlaceholderText } = render(<NewToDoForm />);
  const input = queryByPlaceholderText("Enter new ToDo here...");
  const btn = queryByText("Add New ToDo");
  fireEvent.change(input, { target: { value: "Walk Dog" } });
  fireEvent.click(btn);
  expect(queryByText("Walk Dog")).toBeInTheDocument();
});
