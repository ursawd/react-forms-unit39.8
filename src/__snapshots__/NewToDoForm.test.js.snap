// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      id="task"
      name="task"
      placeholder="Enter new ToDo here..."
      type="Text"
      value=""
    />
    <button
      class="input-btn"
      type="submit"
    >
      Add New ToDo
    </button>
  </form>
</DocumentFragment>
`;
